tools/azure-sdk-tools/packaging_tools/auto_codegen.py

```diff
@@ -37,6 +40,54 @@ def init_new_service(package_name, folder_name):
             with open(str(ci), 'w') as file_out:
                 file_out.writelines(content)
 
+def update_service_metadata(sdk_folder, data, config, folder_name, package_name, spec_folder, input_readme):
+
+    readme_file = str(Path(spec_folder, input_readme))
+    global_conf = config["meta"]
+    local_conf = config["projects"][readme_file]
+
+    cmd = ["autorest", input_readme]
+    cmd += build_autorest_options(global_conf, local_conf)
+
+    # metadata
+    _metadata = {
+        "autorest": global_conf["autorest_options"]["version"],
+        "use": global_conf["autorest_options"]["use"],
+        "commit": data["headSha"],
+        "repository_url": data["repoHttpsUrl"],
+        "autorest_command": " ".join(cmd),
+        "readme": input_readme
+    }
+
+    _LOGGER.info("Metadata json:\n {}".format(json.dumps(_metadata, indent=2)))
+
+    # metadata_folder = Path(sdk_folder, "sdk/metadata/mgmt").expanduser()
+    metadata_folder = Path(sdk_folder, folder_name, package_name).expanduser()
```

Comment:
Change `metadata_folder` to `package_folder` to make the name more straightforward.